src/quoter.ts

```diff
@@ -42,34 +64,35 @@ export abstract class SwapQuoter {
     const singleHop = route.pools.length === 1
     const quoteAmount: string = toHex(amount.quotient)
     let calldata: string
+    const swapInterface = options.useQuoterV2 ? new Interface(IQuoterV2.abi) : this.INTERFACE
 
     if (singleHop) {
-      if (tradeType === TradeType.EXACT_INPUT) {
-        calldata = SwapQuoter.INTERFACE.encodeFunctionData(`quoteExactInputSingle`, [
-          route.tokenPath[0].address,
-          route.tokenPath[1].address,
-          route.pools[0].fee,
-          quoteAmount,
-          toHex(options?.sqrtPriceLimitX96 ?? 0)
-        ])
-      } else {
-        calldata = SwapQuoter.INTERFACE.encodeFunctionData(`quoteExactOutputSingle`, [
-          route.tokenPath[0].address,
-          route.tokenPath[1].address,
-          route.pools[0].fee,
-          quoteAmount,
-          toHex(options?.sqrtPriceLimitX96 ?? 0)
-        ])
+      const quoteV2Params: QuoteParams = {
```

Comment:
this is confusing, should be BaseQuoteParams and each v1/v2 subtype should not have optional amount members - can disambiguate the types down below and add the appropriate amount field depending on the quoter version